river/Cursor.zig

```diff
@@ -637,13 +743,38 @@ pub fn enterMode(self: *Self, mode: std.meta.Tag((Mode)), view: *View) void {
                 .passthrough, .down => unreachable,
                 .move => self.mode = .{ .move = view },
                 .resize => {
-                    const cur_box = &view.current.box;
+                    assert(resize_direction != null);
+
+                    view.setResizing(true);
+
                     self.mode = .{ .resize = .{
                         .view = view,
-                        .offset_x = cur_box.x + @intCast(i32, cur_box.width) - @floatToInt(i32, self.wlr_cursor.x),
-                        .offset_y = cur_box.y + @intCast(i32, cur_box.height) - @floatToInt(i32, self.wlr_cursor.y),
+                        .resize_direction = resize_direction.?,
+                        .offset_x = undefined,
+                        .offset_y = undefined,
                     } };
-                    view.setResizing(true);
+
+                    const cur_box = &view.current.box;
+                    const pointer_surface_x = @floatToInt(i32, self.wlr_cursor.x) - cur_box.x;
+                    const pointer_surface_y = @floatToInt(i32, self.wlr_cursor.y) - cur_box.y;
+                    switch (resize_direction.?) {
+                        .top, .left, .topleft => {
+                            self.mode.resize.offset_x = pointer_surface_x;
+                            self.mode.resize.offset_y = pointer_surface_y;
+                        },
+                        .topright => {
+                            self.mode.resize.offset_x = @intCast(i32, cur_box.width) - pointer_surface_x;
```

Comment:
we should probably either truncate here or not bother with the truncation below and just intCast()